### PR TITLE
[incubator/xray] Bumped Xray version and added RBAC support

### PR DESCRIPTION
**Pull request by jainishshah17**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by jainishshah17** [comment]
@lachie83  Can you please take a look?

**Comment by jainishshah17** [comment]
@eldada

**Comment by jainishshah17** [comment]
@prydonius Any comments?

File: stable/xray/templates/xray-analysis-deployment.yaml [hunk 1]
```diff
@@ -88,7 +88,7 @@ spec:
           mountPath: "/scripts"
       containers:
       - name: {{ .Values.analysis.name }}
-        image: {{ .Values.analysis.image }}:{{ .Values.common.xrayVersion }}
+        image: {{ .Values.analysis.image }}:{{ default .Chart.AppVersion .Values.common.xrayVersion }}
```
Comment (davidkarlsen) [inline stable/xray/templates/xray-analysis-deployment.yaml]: See "Configuration" section in https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md - I think this breaks this.

Comment (jainishshah17) [reply stable/xray/templates/xray-analysis-deployment.yaml]: @davidkarlsen There are couple of charts using this method.

Comment (davidkarlsen) [reply stable/xray/templates/xray-analysis-deployment.yaml]: @unguiculus any viewpoints on this?

Comment (rimusz) [reply stable/xray/templates/xray-analysis-deployment.yaml]: it does not really break the `Chart Review Guidelines` as using `AppVersion` allows better to manage docker tags in multiple places

File: stable/xray/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
+name: xray
+version: 0.3.1
```
Comment (mattfarina) [inline stable/xray/Chart.yaml]: This chart is already in incubator with a version of `0.3.1`. There is a process for promoting a chart from incubator to stable ([documented here](https://github.com/helm/charts/blob/master/PROCESSES.md#promoting-a-chart-from-incubator-to-stable)). Can you follow that and make sure the version in incremented?

The process is designed to help inform existing users figure out the change.

**Review by mattfarina — CHANGES_REQUESTED** [review]

File: stable/xray/Chart.yaml (hunk 2 above)
Comment (jainishshah17) [reply stable/xray/Chart.yaml]: @mattfarina Here is the PR for depreciating incubator chart https://github.com/helm/charts/pull/7156

**Review by mattfarina — APPROVED** [review]

**Pull request merged**